src/casp/services/api/data_manager/cell_operations.py

```diff
@@ -112,52 +119,69 @@ def get_neighborhood_distance_summary_dev_details(
 
         return bigquery_response_parsers.parse_match_query_job(query_job=query_job, include_dev_details=True)
 
-    def get_cell_metadata_by_ids(self, cell_ids: t.List[int]) -> t.List[t.Dict[str, t.Any]]:
+    def _get_cell_metadata_by_ids(
+        self, cell_ids: t.List[int], feature_names: t.List[str]
+    ) -> t.List[schemas.CellariumCellMetadata]:
         """
-        Get cells by ids, maintaining the order of cell_ids. Try to get cell metadata from cache first, then from the
-        database if not found in cache.
+        Get cells by ids, maintaining the order of cell_ids. Preserves the order of cell_ids in the query results.
 
         :param cell_ids: Cas cell indexes to retrieve metadata for.
 
         :return: List of dictionaries representing the query results, ordered according to cell_ids.
         """
-        # Attempt to get cell metadata from cache
-        cache_keys = [f"cell_metadata:{cell_id}" for cell_id in cell_ids]
-        cached_metadata_dicts = self.cache.get_many(cache_keys)
-        found_ids = [int(key.split(":")[-1]) for key in cached_metadata_dicts.keys()]
-        missed_ids = list(set(cell_ids) - set(found_ids))
-
-        # Initialize a dict to hold the final result in order
-        ordered_metadata = {cell_id: None for cell_id in cell_ids}
-
-        # Update the ordered dict with cached results
-        for cell_id in found_ids:
-            ordered_metadata[cell_id] = cached_metadata_dicts[f"cell_metadata:{cell_id}"]
-
-        # If there are missed_ids, retrieve them from the database
-        if missed_ids:
-            columns = [column.key for column in CellInfo.__table__.columns if column.key != "obs_metadata_extra"]
-            with_entity_items = [getattr(CellInfo, name) for name in columns]
-            database_cells = (
-                self.system_data_db_session.query(CellInfo)
-                .with_entities(*with_entity_items)
-                .filter(CellInfo.cas_cell_index.in_(missed_ids))
-                .all()
+        if not cell_ids:
+            raise ValueError("No cell IDs provided")
+
+        if len(cell_ids) > self.MAX_CELL_IDS_PER_QUERY:
+            raise ValueError(f"Number of cell IDs exceeds the maximum of {self.MAX_CELL_IDS_PER_QUERY}")
+
+        with self.system_data_db_session.begin():
+            # Creating temp table
```

Comment:
should we add a comment to specify that the temp table gets auto vacuumed when session ends?